.travis.yml

```diff
@@ -32,31 +34,20 @@ before_install:
 
 install:
  - travis_retry cabal update
- # can't use "cabal install --only-dependencies --enable-tests" due to dep-cycle
- - cabal install
-      "QuickCheck >=2.10 && <2.14"
-      "byteorder ==1.0.*"
-      "dlist >=0.5 && <0.9"
-      "mtl >=2.0 && <2.3"
-      deepseq test-framework-hunit test-framework-quickcheck2
 
 script:
- - cabal configure --enable-tests -v2
- - cabal build
- # --show-details=streaming is available for CABALVER>=1.20 only
- - if [ "$(echo -e "1.20\n$CABALVER" | sort -rV | head -n1)" = "$CABALVER" ]; then
-     cabal test --show-details=streaming;
-   else
-     travis_wait cabal test --show-details=always;
-   fi
- - cabal sdist
+ - cabal v2-configure; cabal v2-build
+ - cabal v2-sdist -o .
+ # Remove to enable build of sdist.
+ - rm -f cabal.project
+ - rm -f tests/test-builder.tix
+ - (cd tests; cabal v2-configure; cabal v2-test)
  # "cabal check" disabled due to -O2 warning
- - export SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}') ;
-   cd dist/;
-   if [ -f "$SRC_TGZ" ]; then
-      cabal install --force-reinstalls "$SRC_TGZ";
+ - export SRC=$(cabal info . | awk '{print $2; exit}');
+   if [ -f "$SRC.tar.gz" ]; then
+      cabal get "./$SRC.tar.gz";
+      (cd "$SRC"; cabal install --force-reinstalls .);
```

Comment:
Did you consider simply using https://github.com/haskell-CI/haskell-ci to generate this script?

Reply:
Actually, that might be tricky…

I can give this a spin once this PR has been merged.

Reply:
I'm having a devil of a time getting it to work portably between Cabal 2.4, 3.0 and 3.2, not sure the autogenerated one would fare better.

Reply:
Got it to work, but had to resort to `v1-install` with Cabal 3.x, if you figure out a working, non-deprecated syntax, I'm all ears...

Reply:
Compatibility issues like that are a `haskell-ci` specialty. ;)

No worries though. Looks like CI is finally green now!